10a-walkthrough-6.Rmd

```diff
@@ -0,0 +1,677 @@
+
+*A note on the gender variable in this dataset*
+
+*This dataset uses a binary approach to data collection about gender. Students are described as either male or female. The need for an inclusive approach to documenting gender identity is discussed in a document by Andrew Park of The Williams Institute at UCLA [authored a document](https://williamsinstitute.law.ucla.edu/wp-content/uploads/Reachable-Data-collection-methods-for-sexual-orientation-gender-identity-March-2016.pdf).*
+
+The gender variables in our dataset are spread across four columns, one representing a combination of gender and age range. We can use `gather` to bring the gender variable into one column. In this transformation, we create two new columns: a `gender` column and a `total` column. The `total` column will contain the number of students in each row's gender and age category.  
+
+```{r}
+child_counts <- child_counts %>% 
+    gather(gender, total, f_3_5:m_6_21)
+```
+
+To make the values of the `gender` column more intuitive, we'll use `case_when` to transform the values to either "f" or "m":
+
+```{r}
+child_counts <- child_counts %>% 
+    mutate(gender = case_when(
+        gender == "f_3_5" ~ "f", 
+        gender == "m_3_5" ~ "m", 
+        gender == "f_6_21" ~ "f", 
+        gender == "m_6_21" ~ "m", 
+        TRUE ~ as.character(gender)
+    ))
+```
+
+## Convert classes
+
+The values in the `total` column represent the number of students from a specific year, state, gender, and age group. We know from the `chr` under their variable names that R is treating these values like characters instead of numbers. While R does a decent job of treating numbers like numbers when needed, it's much safer to prepare the dataset by changing these character columns to number columns. We'll use `dplyr::mutate_at` to conveniently change the count columns. 
+
+```{r convert to numeric}
+child_counts <- child_counts %>% 
+    mutate(total = as.numeric(total)) 
+child_counts
+```
+
+Changing these count columns from character classes to number classes resulted in two changes. First, the `chr` under these variable names has no changed to `dbl`, short for "double-precision". This lets us know that R recognizes these values as numbers with decimal points. Second, the blank values changed to `NA`. When R sees a character class value like `"4"`, it knows to change it to numeric class `4`. But there is no obvious number represented by a character class like `""`, so it changes it to `NA`: 
+
+```{r}
+# Convert a character to a number 
+as.numeric("4")
+
+# Convert a blank character to number
+as.numeric("")
+```
+
+Similarly, the variable `year` needs to be changed from the character format to the date format. Doing so will make sure R treats this variable like a point in time when we plot our dataset. The package `lubridate` has a handy function called `ymd` that can help us. We just have to use the `truncated` argument to let R know we don't have a month and date to convert. 
```

Comment:
I know this chapter is already at about length (and it's an incredible chapter that is featureful), but... :) wonder if we can add a bit more on how this (and related, like `mdy`) lubridate functions work?